core/rs/core/src/tableinfo.rs

```diff
@@ -0,0 +1,261 @@
+use crate::c::CPointer;
+use crate::c::{crsql_ColumnInfo, crsql_TableInfo};
+use alloc::boxed::Box;
+use alloc::ffi::CString;
+use alloc::format;
+use alloc::vec;
+use alloc::vec::Vec;
+use core::ffi::c_char;
+use num_traits::ToPrimitive;
+use sqlite_nostd as sqlite;
+use sqlite_nostd::Connection;
+use sqlite_nostd::ResultCode;
+use sqlite_nostd::StrRef;
+
+/**
+ * Given a table name, return the table info that describes that table.
+ * TableInfo is a struct that represents the results
+ * of pragma_table_info, pragma_index_list, pragma_index_info on a given table
+ * and its indices as well as some extra fields to facilitate crr creation.
+ */
+pub fn pull_table_info(
+    db: *mut sqlite::sqlite3,
+    table: &str,
+    table_info: *mut *mut crsql_TableInfo,
+    err: *mut *mut c_char,
+) -> Result<ResultCode, ResultCode> {
+    let sql = format!("SELECT count(*) FROM pragma_table_info('{table}')");
+    let columns_len = match db.prepare_v2(&sql).and_then(|stmt| {
+        stmt.step()?;
+        stmt.column_int(0)?.to_usize().ok_or(ResultCode::ERROR)
+    }) {
+        Ok(count) => count,
+        Err(code) => {
+            err.set(&format!("Failed to find columns for crr -- {table}"));
```

Comment:
weird that this compiles. You're missing the `table` argument to the format string. Or do format strings just capture whatever variable name happens to be in scope and I've just never tried that? 😅

Reply:
Yeah. It works just like template literals in JS. I find it very convenient.